src/Microsoft.Health.Fhir.Stu3.Core/Features/Definition/CompartmentIndexer.cs

```diff
@@ -70,7 +70,7 @@ public CompartmentIndices Extract(string resourceType, IReadOnlyCollection<Searc
 
             foreach (var indexEntry in searchIndices)
             {
-                if (indexEntry.Value is ReferenceSearchValue refValue && CompartmentDefinitionManager.ResourceTypeToCompartmentType.TryGetValue(refValue.ResourceType, out CompartmentType compartmentType))
+                if (indexEntry.Value is ReferenceSearchValue refValue && !string.IsNullOrWhiteSpace(refValue.ResourceType) && CompartmentDefinitionManager.ResourceTypeToCompartmentType.TryGetValue(refValue.ResourceType, out CompartmentType compartmentType))
```

Comment:
As discussed, please open a bug for when the reference is constrained to a single type. Once we implement referential integrity checking, we will be able to resolve this in all cases.

Reply:
Opened issue #485.